src/ucp/tag/rndv.c

```diff
@@ -525,17 +525,14 @@ static void ucp_rndv_prepare_zcopy(ucp_request_t *sreq, ucp_ep_h ep)
         ucs_assert_always(status == UCS_OK);
     }
 
-    sreq->send.uct_comp.func = ucp_rndv_contig_zcopy_completion;
+    sreq->send.uct_comp.func  = ucp_rndv_contig_zcopy_completion;
+    sreq->send.uct_comp.count = 1;
 
-    max_zcopy = ucp_ep_config(ep)->am.max_zcopy;
-    if (sreq->send.length <= max_zcopy - sizeof(ucp_rndv_data_hdr_t)) {
-        sreq->send.uct_comp.count = 1;
-        sreq->send.uct.func = ucp_rndv_zcopy_single;
+    if (sreq->send.length <= ucp_ep_config(ep)->am.max_zcopy -
+        sizeof(ucp_rndv_data_hdr_t)) {
+        sreq->send.uct.func   = ucp_rndv_zcopy_single;
     } else {
-        /* calculate number of zcopy fragments */
-        sreq->send.uct_comp.count = 1 + (sreq->send.length - 1) /
-                                    (max_zcopy - sizeof(ucp_rndv_data_hdr_t));
-        sreq->send.uct.func = ucp_rndv_zcopy_multi;
+        sreq->send.uct.func   = ucp_rndv_zcopy_multi;
```

Comment:
Can it be so that ucp_do_am_zcopy_multi will need to call uct_ep_am_zcopy several times? For instance when _flag_iov_mid_ flag is true. If yes, then completion count should be set accordingly.